File: app/src/main/java/io/github/wulkanowy/ui/main/MainPresenter.java
package io.github.wulkanowy.ui.main;


import javax.inject.Inject;

import io.github.wulkanowy.data.RepositoryContract;
import io.github.wulkanowy.ui.base.BasePresenter;

public class MainPresenter extends BasePresenter<MainContract.View>
        implements MainContract.Presenter {

    private int currentTabPosition = 0;

    @Inject
    MainPresenter(RepositoryContract repository) {
        super(repository);
    }

    @Override
    public void onTabSelected(int position, boolean wasSelected, int defaultPosition) {
        if (!wasSelected) {
            getView().setChildFragmentSelected(position, true);
            getView().setCurrentPage(position);

            getView().setChildFragmentSelected(position, false);

            currentTabPosition = position;
        }

        if (wasSelected && position == defaultPosition && currentTabPosition == 0) {
            getView().setChildFragmentSelected(position, true);
            currentTabPosition = position;
        }
    }
}

File: app/src/main/java/io/github/wulkanowy/ui/main/MainPresenter.kt
package io.github.wulkanowy.ui.main

import io.github.wulkanowy.data.ErrorHandler
import io.github.wulkanowy.data.repositories.PreferencesRepository
import io.github.wulkanowy.ui.base.BasePresenter
import javax.inject.Inject

class MainPresenter @Inject constructor(
        errorHandler: ErrorHandler,
        private val prefRepository: PreferencesRepository)
    : BasePresenter<MainView>(errorHandler) {

    override fun onAttachView(view: MainView) {
        super.onAttachView(view)
        view.run {
            startMenuIndex = prefRepository.startMenuIndex
            initView()
        }
    }

    fun onStartView() {
        view?.run { setViewTitle(getViewTitle(currentMenuIndex)) }
    }

    fun onMenuViewChange(index: Int) {
        view?.run { setViewTitle(getViewTitle(index)) }
    }

    fun onTabSelected(index: Int, wasSelected: Boolean): Boolean {
        return view?.run {
            if (wasSelected) {
                notifyMenuViewReselected()
                false
            } else {
                switchMenuView(index)
                true
            }
        } == true
    }
}
